spec/acceptance/02_holds_safely_spec.rb

```diff
@@ -0,0 +1,237 @@
+require 'spec_helper_acceptance'
+
+describe 'package_holding_with_apt' do
+  def get_installed_version host, package_name
+    line = on(host, "dpkg -s #{package_name} | grep ^Version").stdout
+    version = line.gsub(/\s+/,'').split(':',2).last
+    version.empty? ? nil : version
+  end
+
+  def get_candidate_version host, package_name
+    line = on(host, "apt-cache policy #{package_name} | grep Candidate: | head -1").stdout
+    version = line.gsub(/\s+/,'').split(':',2).last
+    version.empty? ? nil : version
+  end
+
+  def get_packages_state host
+    packages_state = on(host, 'dpkg-query -W --showformat \'${Status} ${Package}\n\'').stdout
+    packages_state.lines.each_with_object({}) do |line, h|
+      if match = line.match(/^(\S+) +(\S+) +(\S+) (\S+)$/)
+        desired, error, status, name = match.captures
+        h[name] = desired
+      end
+    end
+  end
+
+  def set_package_state host, package, state
+    on(host, "echo #{package} #{state} | dpkg --set-selections")
+  end
+
+  before :all do
+    @managed_packages = [
+      'ubuntu-minimal',
+      'puppetlabs-release-pc1',
+      'puppet-agent',
+      'openssh-server',
+      'dict-jargon',
+      'fortunes',
+    ]
+    @package_versions = {}
+
+    hosts.each do |host|
+      install_package host, 'dict-jargon'
+      expect(check_for_package host, 'dictd').to be true
+      install_package host, 'fortunes'
+      expect(check_for_package host, 'fortunes-min').to be true
+      # same as `include package_purging::config`, saves a Puppet run
+      create_remote_file host, '/etc/apt/apt.conf.d/99always-purge', "APT::Get::Purge \"true\";\n";
```

Comment:
The module provides an `package_purging::config` which is mentioned in the README.

`package_purging::config` is tested in `00_purges_safely_spec.rb`, everywhere else I'm hackishly creating the file by hand to save a couple of Puppet runs. Some context is available [here](https://github.com/hashbrowncipher/puppet-package_purging/pull/4#issuecomment-272715594).